frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.tsx

```diff
@@ -457,46 +500,53 @@ export function FieldValuesWidgetInner({
     function (option: FieldValue): {
       label: string;
       value: string;
+      customLabel?: string;
     } {
       const value = option[0];
       const column = fields[0];
       const label =
+        option[1] ??
         formatValue(value, {
           ...formatOptions,
           column,
           remap: showRemapping(fields),
           jsx: false,
           maximumFractionDigits: 20,
           // we know it is string | number because we are passing jsx: false
-        })?.toString() ?? "<null>";
+        })?.toString() ??
+        "<null>";
 
-      return { value: value?.toString() ?? "", label };
+      return {
+        value: value?.toString() ?? "",
+        label,
+        customLabel: option[1],
+      };
     },
     [fields, formatOptions],
   );
 
   const CustomItemComponent = useMemo(
```

Comment:
The problem is we need to close over some variables from the components scope here. We could use `useCallback` here instead but I think this is safer (since it makes less assumptions about the result).